lib/lib_rss.php

```diff
@@ -507,12 +507,47 @@ function enforceHttpEncoding(string $html, string $contentType = ''): string {
 	return $html;
 }
 
+/**
+ * Set an HTML base URL to the HTML content if there is none.
+ * @param string $html the raw downloaded HTML content
+ * @param string $href the HTML base URL
+ * @return string an HTML string
+ */
+function enforceHtmlBase(string $html, string $href): string {
+	$doc = new DOMDocument();
+	$doc->loadHTML($html, LIBXML_NONET | LIBXML_NOERROR | LIBXML_NOWARNING);
+	if ($doc->documentElement === null) {
+		return '';
+	}
+	$xpath = new DOMXPath($doc);
+	$bases = $xpath->evaluate('//base');
+	if (!($bases instanceof DOMNodeList) || $bases->length === 0) {
+		$base = $doc->createElement('base');
+		if ($base === false) {
+			return $html;
+		}
+		$base->setAttribute('href', $href);
+		$head = null;
+		$heads = $xpath->evaluate('//head');
+		if ($heads instanceof DOMNodeList && $heads->length > 0) {
+			$head = $heads->item(0);
+		}
+		if ($head instanceof DOMElement) {
+			$head->insertBefore($base, $head->firstChild);
+		} else {
+			$doc->insertBefore($base, $doc->documentElement->firstChild);
```

Comment:
There seems to be an error in this code path, changing it to `$doc->documentElement->insertBefore($base, $doc->documentElement->firstChild);` seems to resolve it for me, otherwise it throws a `Fatal error: Uncaught DOMException: Not Found Error`

Reply:
Thanks @dotsam . Would you feel like sending a PR?

Reply:
https://github.com/FreshRSS/FreshRSS/pull/7741